packages/devtools_app/lib/src/screens/vm_developer/object_inspector/vm_ic_data_display.dart

```diff
@@ -82,10 +82,10 @@ class _VmICDataDisplayState extends State<VmICDataDisplay> {
     final entriesFuture = service
         .getObject(isolateId, icData.entries.id!)
         .then((e) => e as Instance);
-    _initialized = Future.wait([
+    _initialized = (
       argumentsDescriptorFuture,
       entriesFuture,
-    ]).then((result) => populateLists(result[0], result[1]));
+    ).wait.then((result) => populateLists(result.$1, result.$2));
```

Comment:
interesting, so `.wait` works on a record too, not just lists? Is there a way to name the record results so that we don't have to use `$1` and `$2` ?

Reply:
> So `.wait` works on a record too, not just lists?

Yep, and that's when it shines as it allows you to access specific items with their respective types :D 

> Is there a way to name the record results so that we don't have to use $1 and $2 ?

The `.wait` extension only exists for records with just positional fields, but you can [destructure](https://dart.dev/language/records#multiple-returns) them using a pattern. In this case, since the function wasn't using async/await, it took a bit more restructuring. I went ahead and did that in https://github.com/flutter/devtools/pull/8619/commits/04334cdef149d95f63ff09aebb03309436263bef :)